x/registry/types/msgs.go

```diff
@@ -0,0 +1,129 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var AllRequestMsgs = []sdk.Msg{
+	(*MsgRegisterNFT)(nil),
+	(*MsgGrantRole)(nil),
+	(*MsgRevokeRole)(nil),
+	(*MsgUnregisterNFT)(nil),
+}
+
+// ValidateBasic validates the MsgRegisterNFT message
+func (m *MsgRegisterNFT) ValidateBasic() error {
+	// Verify the authority is a valid address
+	if _, err := sdk.AccAddressFromBech32(m.Authority); err != nil {
+		return NewErrCodeInvalidField("authority", m.Authority)
+	}
```

Comment:
issue: The `err` here should be included in the return value.

That goes for everything in here. If there's an `err` indicating a problem, it should be included in the returned error. We need to tell people why something is wrong, not just that its wrong.